In [1]:
import pandas as pd

In [2]:
#Cargar los datos
df_respuestas = pd.read_csv('Base Datos - Tesis/t_respuestasnom35.csv', engine='python')
df_empleados = pd.read_csv('Base Datos - Tesis/t_empleadosnom35.csv', engine='python')

# Descubrir y Visualizar los datos para obtener información

In [3]:
#Mostrar las primeras filas
df_respuestas.head()

Id_respuesta  Id_empleado_nom35  Id_res_nom35  respuesta        date
0             1                  1             1          2  2019-06-10
1             2                  1             2          2  2019-06-10
2             3                  1             3          2  2019-06-10
3             4                  1             4          2  2019-06-10
4             5                  1             5          2  2019-06-10

In [4]:
#Mostrar las columnas
df_respuestas.columns.values

array(['Id_respuesta', 'Id_empleado_nom35', 'Id_res_nom35', 'respuesta',
       'date'], dtype=object)

In [5]:
#Mostrar el numero de filas y columnas
df_respuestas.shape

(1265984, 5)

In [6]:
#Mostrar el tipo de variables
df_respuestas.dtypes

Id_respuesta          int64
Id_empleado_nom35     int64
Id_res_nom35          int64
respuesta             int64
date                 object
dtype: object

In [7]:
#Visualizar los distintos tipos de respuestas de la norma 035 dada por empleados
df_respuestas['respuesta'].value_counts()

0    472517
2    317130
1    274435
3    104420
4     97481
6         1
Name: respuesta, dtype: int64

# Visualizacion de los datos de los Empleados

In [8]:
#Mostrar las primeras filas
df_empleados.head

<bound method NDFrame.head of        Id_Empleado                                             Nombre  \
0                1                                      Miguel Orozco   
1                2                                           Ejemplo1   
2                3                                            WangTsu   
3                4                                  Rodrigo Padilla 2   
4                5                                      Miriam Carlos   
...            ...                                                ...   
21119        21247  cb1b15500d18f8f93acb06e3b3779c0754811039a2c169...   
21120        21248  c8ca86bcb1b19554526877d3e8d8e3ddabba11144f35a9...   
21121        21249  c7ce4592cd5bfb617c911352d05f7d1550a0cf5ba6f53d...   
21122        21250  f860de2f24e1f55815a1e53c479d19d781712306cbf4cf...   
21123        21251  83bd54fe16181d08e658a5cb0b4ecc5ca936d67e75d37f...   

                                                   Email        Edad  \
0      e3b0c44298fc1c

In [9]:
#Mostrar las columnas
df_empleados.columns.values

array(['Id_Empleado', 'Nombre', 'Email', 'Edad', 'Genero', 'EstadoCivil',
       'NivelEstudios', 'Ocupacion', 'Departamento', 'TipoPuesto',
       'TipoContratacion', 'TipoPersonal', 'TipoJornada', 'Rotacion',
       'ExperienciaActual', 'ExperienciaTotal', 'PIN', 'Folio',
       'FechaRegistro', 'NombreSucursal', 'CPSucursal', 'Pagado', 'Nota'],
      dtype=object)

In [10]:
#Mostrar el numero de filas y columnas
df_empleados.shape

(21124, 23)

In [11]:
#Mostrar el tipo de variables
df_empleados.dtypes

Id_Empleado           int64
Nombre               object
Email                object
Edad                 object
Genero               object
EstadoCivil          object
NivelEstudios        object
Ocupacion            object
Departamento         object
TipoPuesto           object
TipoContratacion     object
TipoPersonal         object
TipoJornada          object
Rotacion             object
ExperienciaActual    object
ExperienciaTotal     object
PIN                  object
Folio                object
FechaRegistro        object
NombreSucursal       object
CPSucursal           object
Pagado               object
Nota                 object
dtype: object

# Preparar los datos para los algoritmos de Machine Learning

In [12]:
df_filtrado_respuestas=df_respuestas.drop(df_respuestas.index[df_respuestas.Id_empleado_nom35 <= 26], axis=0)
df_filtrado_empleados=df_empleados.drop(df_empleados.index[df_empleados.Id_Empleado <= 26], axis=0)

df_r=pd.DataFrame(df_filtrado_respuestas)
df_e=pd.DataFrame(df_filtrado_empleados)
df_e2=df_e.loc[:, ['Id_Empleado','Genero','EstadoCivil','NivelEstudios','Ocupacion','Departamento','TipoPuesto','TipoContratacion','TipoPersonal','TipoJornada','Rotacion','ExperienciaActual']]
df_r2=df_r.loc[:, :]

In [42]:
#pd.set_option('display.max_columns', None)
#pd.set_option('display.max_rows', None)

In [13]:
df_filtrado_respuestas

Id_respuesta  Id_empleado_nom35  Id_res_nom35  respuesta        date
1007             1008                 28             1          0  2019-07-08
1008             1009                 28             2          2  2019-07-08
1009             1010                 28             3          0  2019-07-08
1010             1011                 28             4          2  2019-07-08
1011             1012                 28             5          2  2019-07-08
...               ...                ...           ...        ...         ...
1265979       1268876              21251           152          2  2021-04-07
1265980       1268877              21251           153          0  2021-04-07
1265981       1268878              21251           154          0  2021-04-07
1265982       1268879              21251           155          3  2021-04-07
1265983       1268880              21251           156          0  2021-04-07

[1264483 rows x 5 columns]

In [14]:
df_filtrado_empleados

Id_Empleado                                             Nombre  \
25              27  64805e8771571d97e53c2bba8d6d29cab52ce2a30274aa...   
26              29  9bab180b309ff8d9e3058c74a0d8f101d35b61d9041319...   
27              30  76ee4ab8668be0081c2e3a4fbf6b2f8f7642aedecbcfa1...   
28              31  b0e186d06d48670968a9c48267b75a69c3f2894147e8e0...   
29              44  733c498ffe3e81d930968947764559d5d0f2a00d8c125e...   
...            ...                                                ...   
21119        21247  cb1b15500d18f8f93acb06e3b3779c0754811039a2c169...   
21120        21248  c8ca86bcb1b19554526877d3e8d8e3ddabba11144f35a9...   
21121        21249  c7ce4592cd5bfb617c911352d05f7d1550a0cf5ba6f53d...   
21122        21250  f860de2f24e1f55815a1e53c479d19d781712306cbf4cf...   
21123        21251  83bd54fe16181d08e658a5cb0b4ecc5ca936d67e75d37f...   

                                                   Email        Edad  \
25     64805e8771571d97e53c2bba8d6d29cab52ce2a30274aa...  1989-02-17   
26     9bab180b309ff8d9e3058c74a0d8f101d35b61d9041319...  1995-08-17   
27     76ee4ab8668be0081c2e3a4fbf6b2f8f7642aedecbcfa1...  1967-03-20   
28     b0e186d06d48670968a9c48267b75a69c3f2894147e8e0...  1979-11-03   
29     733c498ffe3e81d930968947764559d5d0f2a00d8c125e...  1964-10-20   
...                                                  ...         ...   
21119  cb1b15500d18f8f93acb06e3b3779c0754811039a2c169...  1994-08-20   
21120  c8ca86bcb1b19554526877d3e8d8e3ddabba11144f35a9...  1985-12-07   
21121  c7ce4592cd5bfb617c911352d05f7d1550a0cf5ba6f53d...  1984-12-13   
21122  f860de2f24e1f55815a1e53c479d19d781712306cbf4cf...  1996-01-19   
21123  83bd54fe16181d08e658a5cb0b4ecc5ca936d67e75d37f...  1991-09-03   

          Genero EstadoCivil    NivelEstudios            Ocupacion  \
25     Masculino      Casado         Maestria          informatica   
26     Masculino     Soltero     Licenciatura           innovacion   
27     Masculino     Soltero     SinFormacion            direccion   
28     Masculino     Soltero     Licenciatura                otros   
29      Femenino      Casado     Licenciatura            direccion   
...          ...         ...              ...                  ...   
21119   Femenino  UnionLibre         Primaria  PRODUCCIONENPROCESO   
21120   Femenino  UnionLibre       Secundaria  PRODUCCIONENPROCESO   
21121   Femenino  UnionLibre       Secundaria  PRODUCCIONENPROCESO   
21122   Femenino     Soltero     Preparatoria  PRODUCCIONENPROCESO   
21123   Femenino     Soltero  TecnicoSuperior  PRODUCCIONENPROCESO   

              Departamento                              TipoPuesto  ...  \
25             informatica                             profesional  ...   
26              innovacion                                 gerente  ...   
27               direccion                               operativo  ...   
28                   otros                                    otro  ...   
29               direccion                                    otro  ...   
...                    ...                                     ...  ...   
21119  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  ...   
21120  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  ...   
21121  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  ...   
21122  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  ...   
21123  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  ...   

      Rotacion ExperienciaActual ExperienciaTotal     PIN       Folio  \
25          SI              1A4A           10A14A  123abc         NaN   
26          NO              1A4A             6M1A   8u3rq         NaN   
27          SI           Menos6M          Menos6M   ypa5i         NaN   
28          SI              5A9A             5A9A   8bwyp         NaN   
29          NO            15A19A           20A24A   vinxd         NaN   
...        ...               ...              ...     ...         ...   
21119       

In [15]:
#DataFrame para guardar las modificaciones al anterior
#df_e_definitivo = df_e.loc[:, ['TipoPuesto','TipoContratacion','TipoPersonal','TipoJornada','Rotacion','ExperienciaActual']]
df_e

Id_Empleado                                             Nombre  \
25              27  64805e8771571d97e53c2bba8d6d29cab52ce2a30274aa...   
26              29  9bab180b309ff8d9e3058c74a0d8f101d35b61d9041319...   
27              30  76ee4ab8668be0081c2e3a4fbf6b2f8f7642aedecbcfa1...   
28              31  b0e186d06d48670968a9c48267b75a69c3f2894147e8e0...   
29              44  733c498ffe3e81d930968947764559d5d0f2a00d8c125e...   
...            ...                                                ...   
21119        21247  cb1b15500d18f8f93acb06e3b3779c0754811039a2c169...   
21120        21248  c8ca86bcb1b19554526877d3e8d8e3ddabba11144f35a9...   
21121        21249  c7ce4592cd5bfb617c911352d05f7d1550a0cf5ba6f53d...   
21122        21250  f860de2f24e1f55815a1e53c479d19d781712306cbf4cf...   
21123        21251  83bd54fe16181d08e658a5cb0b4ecc5ca936d67e75d37f...   

                                                   Email        Edad  \
25     64805e8771571d97e53c2bba8d6d29cab52ce2a30274aa...  1989-02-17   
26     9bab180b309ff8d9e3058c74a0d8f101d35b61d9041319...  1995-08-17   
27     76ee4ab8668be0081c2e3a4fbf6b2f8f7642aedecbcfa1...  1967-03-20   
28     b0e186d06d48670968a9c48267b75a69c3f2894147e8e0...  1979-11-03   
29     733c498ffe3e81d930968947764559d5d0f2a00d8c125e...  1964-10-20   
...                                                  ...         ...   
21119  cb1b15500d18f8f93acb06e3b3779c0754811039a2c169...  1994-08-20   
21120  c8ca86bcb1b19554526877d3e8d8e3ddabba11144f35a9...  1985-12-07   
21121  c7ce4592cd5bfb617c911352d05f7d1550a0cf5ba6f53d...  1984-12-13   
21122  f860de2f24e1f55815a1e53c479d19d781712306cbf4cf...  1996-01-19   
21123  83bd54fe16181d08e658a5cb0b4ecc5ca936d67e75d37f...  1991-09-03   

          Genero EstadoCivil    NivelEstudios            Ocupacion  \
25     Masculino      Casado         Maestria          informatica   
26     Masculino     Soltero     Licenciatura           innovacion   
27     Masculino     Soltero     SinFormacion            direccion   
28     Masculino     Soltero     Licenciatura                otros   
29      Femenino      Casado     Licenciatura            direccion   
...          ...         ...              ...                  ...   
21119   Femenino  UnionLibre         Primaria  PRODUCCIONENPROCESO   
21120   Femenino  UnionLibre       Secundaria  PRODUCCIONENPROCESO   
21121   Femenino  UnionLibre       Secundaria  PRODUCCIONENPROCESO   
21122   Femenino     Soltero     Preparatoria  PRODUCCIONENPROCESO   
21123   Femenino     Soltero  TecnicoSuperior  PRODUCCIONENPROCESO   

              Departamento                              TipoPuesto  ...  \
25             informatica                             profesional  ...   
26              innovacion                                 gerente  ...   
27               direccion                               operativo  ...   
28                   otros                                    otro  ...   
29               direccion                                    otro  ...   
...                    ...                                     ...  ...   
21119  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  ...   
21120  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  ...   
21121  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  ...   
21122  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  ...   
21123  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  ...   

      Rotacion ExperienciaActual ExperienciaTotal     PIN       Folio  \
25          SI              1A4A           10A14A  123abc         NaN   
26          NO              1A4A             6M1A   8u3rq         NaN   
27          SI           Menos6M          Menos6M   ypa5i         NaN   
28          SI              5A9A             5A9A   8bwyp         NaN   
29          NO            15A19A           20A24A   vinxd         NaN   
...        ...               ...              ...     ...         ...   
21119       

# Saber los distintos datos de cada columna

In [16]:
df_e2['EstadoCivil'].value_counts()

Soltero       9113
Casado        8353
UnionLibre    2816
Divorciado     699
Viudo          118
Name: EstadoCivil, dtype: int64

In [17]:
df_e2['NivelEstudios'].value_counts()

Licenciatura       7992
Secundaria         3172
Preparatoria       3070
Bachillerato       2166
TecnicoSuperior    1707
SinFormacion       1395
Maestria           1044
Primaria            501
Doctorado            52
Name: NivelEstudios, dtype: int64

In [18]:
df_e2['Ocupacion'].value_counts()

produccion                     2609
otros                          2233
ventas                         1560
direccion                      1548
administracion                 1326
                               ... 
700GerenciaGeneral                1
777Saltillo                       1
AssetReliabilityPerformance       1
TechnicalServices                 1
MEJORACONTINUA                    1
Name: Ocupacion, Length: 308, dtype: int64

In [19]:
df_e2['Departamento'].value_counts()

produccion                     2609
otros                          2233
ventas                         1560
direccion                      1548
administracion                 1326
                               ... 
700GerenciaGeneral                1
777Saltillo                       1
AssetReliabilityPerformance       1
TechnicalServices                 1
MEJORACONTINUA                    1
Name: Departamento, Length: 308, dtype: int64

In [20]:
df_e2['TipoPuesto'].value_counts()

operativo                          6552
profesional                        2674
otro                               2508
gerente                            1207
supervisor                         1197
                                   ... 
GerentedeRecursosHumanosExterno       1
SUBDIRECTORIFRS9                      1
EjecutivodeImplementaciones           1
GTIFRS69                              1
VELADOR                               1
Name: TipoPuesto, Length: 1204, dtype: int64

In [21]:
df_e2['TipoContratacion'].value_counts()

tiempoindeterminado    16652
porObraoProyecto        2113
tiempodeterminado       1393
otro                     689
honorarios               252
Name: TipoContratacion, dtype: int64

In [22]:
df_e2['TipoPersonal'].value_counts()

confianza        8497
ninguno          6360
sindicalizado    6242
Name: TipoPersonal, dtype: int64

In [23]:
df_e2['TipoJornada'].value_counts()

diurno      16587
mixto        4035
nocturno      477
Name: TipoJornada, dtype: int64

In [24]:
df_e2['Rotacion'].value_counts()

NO    13750
SI     7349
Name: Rotacion, dtype: int64

In [25]:
df_e2['ExperienciaActual'].value_counts()

1A4A       7959
Menos6M    3969
5A9A       3312
6M1A       2974
10A14A     1451
15A19A      692
20A24A      387
Mas25A      355
Name: ExperienciaActual, dtype: int64

In [26]:
#DataFrame para guardar las modificaciones al anterior de los atribuos que si merecen la pena cambiar sus valores
df_e = df_e.loc[:, ['Id_Empleado','Genero','EstadoCivil','NivelEstudios','Ocupacion','Departamento','TipoPuesto','TipoContratacion','TipoPersonal','TipoJornada','Rotacion','ExperienciaActual']]
df_e

Id_Empleado     Genero EstadoCivil    NivelEstudios  \
25              27  Masculino      Casado         Maestria   
26              29  Masculino     Soltero     Licenciatura   
27              30  Masculino     Soltero     SinFormacion   
28              31  Masculino     Soltero     Licenciatura   
29              44   Femenino      Casado     Licenciatura   
...            ...        ...         ...              ...   
21119        21247   Femenino  UnionLibre         Primaria   
21120        21248   Femenino  UnionLibre       Secundaria   
21121        21249   Femenino  UnionLibre       Secundaria   
21122        21250   Femenino     Soltero     Preparatoria   
21123        21251   Femenino     Soltero  TecnicoSuperior   

                 Ocupacion         Departamento  \
25             informatica          informatica   
26              innovacion           innovacion   
27               direccion            direccion   
28                   otros                otros   
29               direccion            direccion   
...                    ...                  ...   
21119  PRODUCCIONENPROCESO  PRODUCCIONENPROCESO   
21120  PRODUCCIONENPROCESO  PRODUCCIONENPROCESO   
21121  PRODUCCIONENPROCESO  PRODUCCIONENPROCESO   
21122  PRODUCCIONENPROCESO  PRODUCCIONENPROCESO   
21123  PRODUCCIONENPROCESO  PRODUCCIONENPROCESO   

                                   TipoPuesto     TipoContratacion  \
25                                profesional  tiempoindeterminado   
26                                    gerente  tiempoindeterminado   
27                                  operativo     porObraoProyecto   
28                                       otro  tiempoindeterminado   
29                                       otro                 otro   
...                                       ...                  ...   
21119  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  tiempoindeterminado   
21120  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  tiempoindeterminado   
21121  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  tiempoindeterminado   
21122  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  tiempoindeterminado   
21123  AYUDANTEGENERALPARAPRODUCCIONENPROCESO  tiempoindeterminado   

        TipoPersonal TipoJornada Rotacion ExperienciaActual  
25     sindicalizado      diurno       SI              1A4A  
26     sindicalizado      diurno       NO              1A4A  
27     sindicalizado      diurno       SI           Menos6M  
28           ninguno      diurno       SI              5A9A  
29         confianza      diurno       NO            15A19A  
...              ...         ...      ...               ...  
21119        ninguno      diurno       NO              1A4A  
21120        ninguno      diurno       NO              6M1A  
21121        ninguno      diurno       NO              6M1A  
21122        ninguno      diurno       NO           Menos6M  
21123        ninguno      diurno       NO              5A9A  

[21099 rows x 12 columns]

In [27]:
#El 1 representa al genero masculino y el 0 al genero femenino
gen_ = {'Masculino':1, 'Femenino':0}
df_e['Genero'] = df_e['Genero'].map(gen_)
df_e

Id_Empleado  Genero EstadoCivil    NivelEstudios            Ocupacion  \
25              27       1      Casado         Maestria          informatica   
26              29       1     Soltero     Licenciatura           innovacion   
27              30       1     Soltero     SinFormacion            direccion   
28              31       1     Soltero     Licenciatura                otros   
29              44       0      Casado     Licenciatura            direccion   
...            ...     ...         ...              ...                  ...   
21119        21247       0  UnionLibre         Primaria  PRODUCCIONENPROCESO   
21120        21248       0  UnionLibre       Secundaria  PRODUCCIONENPROCESO   
21121        21249       0  UnionLibre       Secundaria  PRODUCCIONENPROCESO   
21122        21250       0     Soltero     Preparatoria  PRODUCCIONENPROCESO   
21123        21251       0     Soltero  TecnicoSuperior  PRODUCCIONENPROCESO   

              Departamento                              TipoPuesto  \
25             informatica                             profesional   
26              innovacion                                 gerente   
27               direccion                               operativo   
28                   otros                                    otro   
29               direccion                                    otro   
...                    ...                                     ...   
21119  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21120  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21121  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21122  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21123  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   

          TipoContratacion   TipoPersonal TipoJornada Rotacion  \
25     tiempoindeterminado  sindicalizado      diurno       SI   
26     tiempoindeterminado  sindicalizado      diurno       NO   
27        porObraoProyecto  sindicalizado      diurno       SI   
28     tiempoindeterminado        ninguno      diurno       SI   
29                    otro      confianza      diurno       NO   
...                    ...            ...         ...      ...   
21119  tiempoindeterminado        ninguno      diurno       NO   
21120  tiempoindeterminado        ninguno      diurno       NO   
21121  tiempoindeterminado        ninguno      diurno       NO   
21122  tiempoindeterminado        ninguno      diurno       NO   
21123  tiempoindeterminado        ninguno      diurno       NO   

      ExperienciaActual  
25                 1A4A  
26                 1A4A  
27              Menos6M  
28                 5A9A  
29               15A19A  
...                 ...  
21119              1A4A  
21120              6M1A  
21121              6M1A  
21122           Menos6M  
21123              5A9A  

[21099 rows x 12 columns]

In [28]:
estudio_ = {'Licenciatura':9, 'Secundaria':8, 'Preparatoria':7, 'Bachillerato':6, 'TecnicoSuperior':5, 'SinFormacion':4, 'Maestria':3, 'Primaria':2, 'Doctorado':1}
df_e['NivelEstudios'] = df_e['NivelEstudios'].map(estudio_)
df_e

Id_Empleado  Genero EstadoCivil  NivelEstudios            Ocupacion  \
25              27       1      Casado              3          informatica   
26              29       1     Soltero              9           innovacion   
27              30       1     Soltero              4            direccion   
28              31       1     Soltero              9                otros   
29              44       0      Casado              9            direccion   
...            ...     ...         ...            ...                  ...   
21119        21247       0  UnionLibre              2  PRODUCCIONENPROCESO   
21120        21248       0  UnionLibre              8  PRODUCCIONENPROCESO   
21121        21249       0  UnionLibre              8  PRODUCCIONENPROCESO   
21122        21250       0     Soltero              7  PRODUCCIONENPROCESO   
21123        21251       0     Soltero              5  PRODUCCIONENPROCESO   

              Departamento                              TipoPuesto  \
25             informatica                             profesional   
26              innovacion                                 gerente   
27               direccion                               operativo   
28                   otros                                    otro   
29               direccion                                    otro   
...                    ...                                     ...   
21119  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21120  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21121  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21122  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21123  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   

          TipoContratacion   TipoPersonal TipoJornada Rotacion  \
25     tiempoindeterminado  sindicalizado      diurno       SI   
26     tiempoindeterminado  sindicalizado      diurno       NO   
27        porObraoProyecto  sindicalizado      diurno       SI   
28     tiempoindeterminado        ninguno      diurno       SI   
29                    otro      confianza      diurno       NO   
...                    ...            ...         ...      ...   
21119  tiempoindeterminado        ninguno      diurno       NO   
21120  tiempoindeterminado        ninguno      diurno       NO   
21121  tiempoindeterminado        ninguno      diurno       NO   
21122  tiempoindeterminado        ninguno      diurno       NO   
21123  tiempoindeterminado        ninguno      diurno       NO   

      ExperienciaActual  
25                 1A4A  
26                 1A4A  
27              Menos6M  
28                 5A9A  
29               15A19A  
...                 ...  
21119              1A4A  
21120              6M1A  
21121              6M1A  
21122           Menos6M  
21123              5A9A  

[21099 rows x 12 columns]

In [29]:
contratacion_ = {'tiempoindeterminado':5,'porObraoProyecto':4,'tiempodeterminado':3,'otro':2,'honorarios':1}
df_e['TipoContratacion'] = df_e['TipoContratacion'].map(contratacion_)
df_e

Id_Empleado  Genero EstadoCivil  NivelEstudios            Ocupacion  \
25              27       1      Casado              3          informatica   
26              29       1     Soltero              9           innovacion   
27              30       1     Soltero              4            direccion   
28              31       1     Soltero              9                otros   
29              44       0      Casado              9            direccion   
...            ...     ...         ...            ...                  ...   
21119        21247       0  UnionLibre              2  PRODUCCIONENPROCESO   
21120        21248       0  UnionLibre              8  PRODUCCIONENPROCESO   
21121        21249       0  UnionLibre              8  PRODUCCIONENPROCESO   
21122        21250       0     Soltero              7  PRODUCCIONENPROCESO   
21123        21251       0     Soltero              5  PRODUCCIONENPROCESO   

              Departamento                              TipoPuesto  \
25             informatica                             profesional   
26              innovacion                                 gerente   
27               direccion                               operativo   
28                   otros                                    otro   
29               direccion                                    otro   
...                    ...                                     ...   
21119  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21120  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21121  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21122  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21123  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   

       TipoContratacion   TipoPersonal TipoJornada Rotacion ExperienciaActual  
25                    5  sindicalizado      diurno       SI              1A4A  
26                    5  sindicalizado      diurno       NO              1A4A  
27                    4  sindicalizado      diurno       SI           Menos6M  
28                    5        ninguno      diurno       SI              5A9A  
29                    2      confianza      diurno       NO            15A19A  
...                 ...            ...         ...      ...               ...  
21119                 5        ninguno      diurno       NO              1A4A  
21120                 5        ninguno      diurno       NO              6M1A  
21121                 5        ninguno      diurno       NO              6M1A  
21122                 5        ninguno      diurno       NO           Menos6M  
21123                 5        ninguno      diurno       NO              5A9A  

[21099 rows x 12 columns]

In [30]:
tipoPersonal_ = {'confianza':3,'ninguno':2,'sindicalizado':1}
df_e['TipoPersonal'] = df_e['TipoPersonal'].map(tipoPersonal_)
df_e

Id_Empleado  Genero EstadoCivil  NivelEstudios            Ocupacion  \
25              27       1      Casado              3          informatica   
26              29       1     Soltero              9           innovacion   
27              30       1     Soltero              4            direccion   
28              31       1     Soltero              9                otros   
29              44       0      Casado              9            direccion   
...            ...     ...         ...            ...                  ...   
21119        21247       0  UnionLibre              2  PRODUCCIONENPROCESO   
21120        21248       0  UnionLibre              8  PRODUCCIONENPROCESO   
21121        21249       0  UnionLibre              8  PRODUCCIONENPROCESO   
21122        21250       0     Soltero              7  PRODUCCIONENPROCESO   
21123        21251       0     Soltero              5  PRODUCCIONENPROCESO   

              Departamento                              TipoPuesto  \
25             informatica                             profesional   
26              innovacion                                 gerente   
27               direccion                               operativo   
28                   otros                                    otro   
29               direccion                                    otro   
...                    ...                                     ...   
21119  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21120  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21121  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21122  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21123  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   

       TipoContratacion  TipoPersonal TipoJornada Rotacion ExperienciaActual  
25                    5             1      diurno       SI              1A4A  
26                    5             1      diurno       NO              1A4A  
27                    4             1      diurno       SI           Menos6M  
28                    5             2      diurno       SI              5A9A  
29                    2             3      diurno       NO            15A19A  
...                 ...           ...         ...      ...               ...  
21119                 5             2      diurno       NO              1A4A  
21120                 5             2      diurno       NO              6M1A  
21121                 5             2      diurno       NO              6M1A  
21122                 5             2      diurno       NO           Menos6M  
21123                 5             2      diurno       NO              5A9A  

[21099 rows x 12 columns]

In [31]:
tipoJornada_ = {'diurno':3,'mixto':2,'nocturno':1}
df_e['TipoJornada'] = df_e['TipoJornada'].map(tipoJornada_)
df_e

Id_Empleado  Genero EstadoCivil  NivelEstudios            Ocupacion  \
25              27       1      Casado              3          informatica   
26              29       1     Soltero              9           innovacion   
27              30       1     Soltero              4            direccion   
28              31       1     Soltero              9                otros   
29              44       0      Casado              9            direccion   
...            ...     ...         ...            ...                  ...   
21119        21247       0  UnionLibre              2  PRODUCCIONENPROCESO   
21120        21248       0  UnionLibre              8  PRODUCCIONENPROCESO   
21121        21249       0  UnionLibre              8  PRODUCCIONENPROCESO   
21122        21250       0     Soltero              7  PRODUCCIONENPROCESO   
21123        21251       0     Soltero              5  PRODUCCIONENPROCESO   

              Departamento                              TipoPuesto  \
25             informatica                             profesional   
26              innovacion                                 gerente   
27               direccion                               operativo   
28                   otros                                    otro   
29               direccion                                    otro   
...                    ...                                     ...   
21119  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21120  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21121  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21122  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21123  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   

       TipoContratacion  TipoPersonal  TipoJornada Rotacion ExperienciaActual  
25                    5             1            3       SI              1A4A  
26                    5             1            3       NO              1A4A  
27                    4             1            3       SI           Menos6M  
28                    5             2            3       SI              5A9A  
29                    2             3            3       NO            15A19A  
...                 ...           ...          ...      ...               ...  
21119                 5             2            3       NO              1A4A  
21120                 5             2            3       NO              6M1A  
21121                 5             2            3       NO              6M1A  
21122                 5             2            3       NO           Menos6M  
21123                 5             2            3       NO              5A9A  

[21099 rows x 12 columns]

In [32]:
rotacion_ = {'NO':2,'SI':1}
df_e['Rotacion'] = df_e['Rotacion'].map(rotacion_)
df_e

Id_Empleado  Genero EstadoCivil  NivelEstudios            Ocupacion  \
25              27       1      Casado              3          informatica   
26              29       1     Soltero              9           innovacion   
27              30       1     Soltero              4            direccion   
28              31       1     Soltero              9                otros   
29              44       0      Casado              9            direccion   
...            ...     ...         ...            ...                  ...   
21119        21247       0  UnionLibre              2  PRODUCCIONENPROCESO   
21120        21248       0  UnionLibre              8  PRODUCCIONENPROCESO   
21121        21249       0  UnionLibre              8  PRODUCCIONENPROCESO   
21122        21250       0     Soltero              7  PRODUCCIONENPROCESO   
21123        21251       0     Soltero              5  PRODUCCIONENPROCESO   

              Departamento                              TipoPuesto  \
25             informatica                             profesional   
26              innovacion                                 gerente   
27               direccion                               operativo   
28                   otros                                    otro   
29               direccion                                    otro   
...                    ...                                     ...   
21119  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21120  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21121  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21122  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21123  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   

       TipoContratacion  TipoPersonal  TipoJornada  Rotacion ExperienciaActual  
25                    5             1            3         1              1A4A  
26                    5             1            3         2              1A4A  
27                    4             1            3         1           Menos6M  
28                    5             2            3         1              5A9A  
29                    2             3            3         2            15A19A  
...                 ...           ...          ...       ...               ...  
21119                 5             2            3         2              1A4A  
21120                 5             2            3         2              6M1A  
21121                 5             2            3         2              6M1A  
21122                 5             2            3         2           Menos6M  
21123                 5             2            3         2              5A9A  

[21099 rows x 12 columns]

In [33]:
experiencia_ = {'1A4A':8,'Menos6M':7,'5A9A':6,'6M1A':5,'10A14A':4,'15A19A':3,'20A24A':2,'Mas25A':1}
df_e['ExperienciaActual'] = df_e['ExperienciaActual'].map(experiencia_)
df_e

Id_Empleado  Genero EstadoCivil  NivelEstudios            Ocupacion  \
25              27       1      Casado              3          informatica   
26              29       1     Soltero              9           innovacion   
27              30       1     Soltero              4            direccion   
28              31       1     Soltero              9                otros   
29              44       0      Casado              9            direccion   
...            ...     ...         ...            ...                  ...   
21119        21247       0  UnionLibre              2  PRODUCCIONENPROCESO   
21120        21248       0  UnionLibre              8  PRODUCCIONENPROCESO   
21121        21249       0  UnionLibre              8  PRODUCCIONENPROCESO   
21122        21250       0     Soltero              7  PRODUCCIONENPROCESO   
21123        21251       0     Soltero              5  PRODUCCIONENPROCESO   

              Departamento                              TipoPuesto  \
25             informatica                             profesional   
26              innovacion                                 gerente   
27               direccion                               operativo   
28                   otros                                    otro   
29               direccion                                    otro   
...                    ...                                     ...   
21119  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21120  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21121  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21122  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   
21123  PRODUCCIONENPROCESO  AYUDANTEGENERALPARAPRODUCCIONENPROCESO   

       TipoContratacion  TipoPersonal  TipoJornada  Rotacion  \
25                    5             1            3         1   
26                    5             1            3         2   
27                    4             1            3         1   
28                    5             2            3         1   
29                    2             3            3         2   
...                 ...           ...          ...       ...   
21119                 5             2            3         2   
21120                 5             2            3         2   
21121                 5             2            3         2   
21122                 5             2            3         2   
21123                 5             2            3         2   

       ExperienciaActual  
25                     8  
26                     8  
27                     7  
28                     6  
29                     3  
...                  ...  
21119                  8  
21120                  5  
21121                  5  
21122                  7  
21123                  6  

[21099 rows x 12 columns]

In [34]:
count=0

#Saber la cantidad de preguntas que respondio cada persona
df_r['Id_empleado_nom35'].value_counts()

15615    136
625      136
5138     118
312      118
5233     115
        ... 
2688      40
2747      40
2753      40
2760      40
28        40
Name: Id_empleado_nom35, Length: 21124, dtype: int64

Hubo personas que solo tuvieron que contestar 40 preguntas, pero huvo el caso de algunos que fue necesario responder hasta 136 preguntas. Esto se debe porque a la hora de responde la norma 035 y al terminar una seccion te dice que dado algunas respuestas anteriores podras finalizar ya las preguntas o deberas seguir contestandolas

In [35]:
#Antes de compilar saber cuantos personas contestaron la encuesta, saber la cantidad de numeros distintos
contador=25
for i in range(0,21124):
    df_prueba = df_r
    df_prueba=df_prueba.drop(df_prueba.index[df_prueba.Id_empleado_nom35 != 27], axis=0)
    df_e.loc[contador, 'Resultado']=df_prueba['respuesta'].sum()
    count+=1
    contador+=1
    #df_prueba['respuesta'].sum()

df_prueba

Id_respuesta  Id_empleado_nom35  Id_res_nom35  respuesta        date
1047          1048                 27             1          2  2019-07-08
1048          1049                 27             2          2  2019-07-08
1049          1050                 27             3          2  2019-07-08
1050          1051                 27             4          2  2019-07-08
1051          1052                 27             5          2  2019-07-08
1052          1053                 27             6          2  2019-07-08
1053          1054                 27             7          2  2019-07-08
1054          1055                 27             8          2  2019-07-08
1055          1056                 27             9          2  2019-07-08
1056          1057                 27            10          2  2019-07-08
1057          1058                 27            11          2  2019-07-08
1058          1059                 27            12          2  2019-07-08
1059          1060                 27            13          2  2019-07-08
1060          1061                 27            14          2  2019-07-08
1061          1062                 27            15          2  2019-07-08
1062          1063                 27            16          2  2019-07-08
1063          1064                 27            17          2  2019-07-08
1064          1065                 27            18          2  2019-07-08
1065          1066                 27            19          2  2019-07-08
1066          1067                 27            20          2  2019-07-08
1067          1068                 27            21          2  2019-07-08
1068          1069                 27            22          2  2019-07-08
1069          1070                 27            23          2  2019-07-08
1070          1071                 27            24          2  2019-07-08
1071          1072                 27            25          2  2019-07-08
1072          1073                 27            26          2  2019-07-08
1073          1074                 27            27          2  2019-07-08
1074          1075                 27            28          2  2019-07-08
1075          1076                 27            29          2  2019-07-08
1076          1077                 27            30          2  2019-07-08
1077          1078                 27            31          2  2019-07-08
1078          1079                 27            32          2  2019-07-08
1079          1080                 27            33          2  2019-07-08
1080          1081                 27            34          2  2019-07-08
1081          1082                 27            35          2  2019-07-08
1082          1083                 27            36          2  2019-07-08
1083          1084                 27            37          2  2019-07-08
1084          1085                 27            38          2  2019-07-08
1085          1086                 27            39          2  2019-07-08
1086          1087                 27            40          2  2019-07-08
1087          1088                 27            41          2  2019-07-08
1088          1089                 27            42          2  2019-07-08
1089          1090                 27            43          2  2019-07-08
1090          1091                 27            44          2  2019-07-08
1091          1092                 27            45          2  2019-07-08
1092          1093                 27            46          2  2019-07-08